sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,420 @@
+tool.  This tool creates a binary that contains individual sections that are
+labeled with `__CLANG_OFFLOAD_BUNDLE__kind-triple`.  This format does not
+match the new format as described above.  To enable an ease of transition
+from the old model to the new model, support will be added that allows for
+support of identifying and extracting device binaries that are using the old
+model.
+
+It is expected that all new binaries generated with the updated offloading
+model will represent the embedded fat object format, moving away from the
+`clang-offload-bundler` usage.  We will not support a mixing of fat object
+formats such as objects created with the bundler combined with embedded objects
+in a single archive or objects created with an embedded offload section and
+additional sections generated with the bundler.
+
+### Device Linking
+
+During this phase, all of the individual device binaries that are extracted and
+are associated with a given target are worked on given the type of binary we
+are working with.  The default device code is typically represented in LLVM-IR
+which requires an additional link step of the device code before being
+wrapped and integrated into the final executable.  As mentioned in
+[Packager](#packager) device representation in SPIR-V should be considered
+with the ability to link native SPIR-V modules.  Performing the device link
+against LLVM-IR is performed by `llvm-link`.
+
+To reduce the potential size of the linked device binary, an additional host
+link step is performed to gather dependency information when static device
+libraries are being compiled.  This information is sent through the
+`clang-offload-deps` tool to generate a dependency IR file which is used
+during the device link step.
+
+The use of the clang-linker-wrapper introduces the support of LTO for
+device code.  We can leverage this and move away from the dependency gathering
+information step with 'clang-offload-deps' and use thinLTO for device code.
+
+There are multiple device linking steps that also occur.  The first step links
+together all of the objects and the required device libraries.  The second is
+performed including all of optional device libraries, the static device
+libraries and the dependency information that was gathered above.  This link
+step is performed with `--only-needed` to streamline the final device binary.
+
+A list of device libraries that need to linked in with user code is provided by
+the driver.  The driver is also responsible for letting the
+`clang-linker-wrapper` know the location of the device libraries.
+
+|            Option                    |   Expected Behavior   |
+|--------------------------------------|-----------------------|
+| `--sycl-device-libraries=<arg>`      | A comma separated list of device libraries that are linked during the device link. |
+| `--device-library-location=<arg>`    | The location in which the device libraries reside |
```

Comment:
Sorry. Missd this earlier. Can we pls name this --sycl-device-library-location ?

Thanks

Reply:
Done.